internal/compiler/object_tree.rs

```diff
@@ -38,10 +38,13 @@ macro_rules! unwrap_or_continue {
     };
 }
 
+pub type DocumentVersion = Option<i32>;
+
 /// The full document (a complete file)
 #[derive(Default, Debug)]
 pub struct Document {
     pub node: Option<syntax_nodes::Document>,
+    pub version: DocumentVersion,
```

Comment:
Idea: Instead of having the version in there, it could be in the SourceFile (SuorceFileInner)  accessible via the Document::node
I thin kit would make everything much simpler, wouldn't it?